### ISSUE TITLE
Add an option for hbs files to be considered route templates and compiled as such

### ISSUE THREAD

**Issue by balinterdi** [embroider-build/embroider#1776]
### Problem description

(I'm taking up the thread from #1756 as requested by @mansona during an Embroider Initiative call.)

Currently, all `hbs` files in v2 addons are considered component templates. If there's no co-located `.js` file, [they will be considered a template-only component](https://github.com/embroider-build/embroider/blob/289f83d80d133da3de2b342e63f937986248c07d/packages/addon-dev/src/rollup-hbs-plugin.ts#L79-L88), otherwise `setComponentTemplate` will be called with the template content and the component class.

This prevents using *route templates* (which are also `.hbs`) in v2 addons but, at least in the initial, migration period there exist apps which would need that feature (the need came from an actual client project).

On the longer term, using a gjs [`ember-route-template`](https://github.com/discourse/ember-route-template) elegantly side-steps that problem (I have a small, working example [here](https://github.com/balinterdi/app-files-from-v2-addons/tree/use-ember-route-template-to-make-route-template-available)), but since it requires the template to be in strict-mode, migrating existing route templates to ember-route-templates is quite an undertaking.

In #1756 I provided a working modification to Embroider and a simple Rollup plugin which allowed compiling route templates. However, it had a few drawbacks:

- It had a hard-wired path to a folder in which templates are considered route templates
- It produced a built output which was "wrong" in the sense that it contained a line that made no sense (see the `setComponentTemplate` call in the below code snippet), but it still worked!

```
import { precompileTemplate } from '@ember/template-compilation';
import { setComponentTemplate } from '@ember/component';

var TEMPLATE = precompileTemplate("The Caro-Kann Opening\n\n<div class=\"image-container\">\n  <img src=\"/grand-prix/assets/images/caro-kann.webp\" alt=\"Caro Kann Opening\" />\n</div>");

var caroKann = setComponentTemplate(TEMPLATE, precompileTemplate("The Caro-Kann Opening\n\n<div class=\"image-container\">\n  <img src=\"/grand-prix/assets/images/caro-kann.webp\" alt=\"Caro Kann Opening\" />\n</div>"));

export { caroKann as default };
```

### Possible solution

In the same PR, [@ef4 suggested](https://github.com/embroider-build/embroider/pull/1756#issuecomment-1895912346) to allow "people "o provide globs for including or excluding files from being treated as template-only components.", which I think is a great idea.

On top of that, we'd need to prevent [the Babel template-colocation-plugin](https://github.com/embroider-build/embroider/blob/main/packages/shared-internals/src/template-colocation-plugin.ts) from inserting the `setComponentTemplate` call in the output. As said above, it doesn't break the template compilation but it makes no sense, and it's superfluous.

**Comment by balinterdi** [comment]
@mansona I saw the Mainmatter Embroider team, and then lolmaus being assigned and then unassigned on the issue. Is this being worked on? Is there something I can do to help?

**Comment by BlueCutOfficial** [comment]
PR #1856 is currently a draft that allows the compilation of hbs route templates as simple templates instead of template-only components. I still have a couple of tests to run to clarify the constraints on the template.

**Comment by balinterdi** [comment]
@BlueCutOfficial Thank you so much, Marine, that looks good!

**Comment by mansona** [comment]
This is fixed in @embroider/addon-dev 4.3.0 👍

**Comment by balinterdi** [comment]
@BlueCutOfficial @mansona Thanks so much!
